#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "tarkbot_robot"

gen = ParameterGenerator()

gen.add("imu_calibrate", bool_t, 0, "Check to start the calibration", False)

gen.add("light_calibrate", bool_t, 0, "save the change about light", False)

light_chose = gen.enum([gen.const("m1",int_t,0x01,"m1"),  
              gen.const("m2",int_t,0x02,"m2"),
              gen.const("m3",int_t,0x03,"m3"),
              gen.const("m4",int_t,0x04,"m4"),
              gen.const("m5",int_t,0x05,"m5"),
              gen.const("m6",int_t,0x06,"m6")],"light_chose")

gen.add("light_chose",int_t,0,"alternative choices",0x02,0x01,0x06,edit_method=light_chose) 

gen.add("RGB_R",int_t,0,"int RGB_R",0,0,255)
gen.add("RGB_G",int_t,0,"int RGB_G",255,0,255)
gen.add("RGB_B",int_t,0,"int RGB_B",0,0,255)

exit(gen.generate(PACKAGE, "tarkbot_robot", "robot"))
